src/components/ResourceList.tsx

```diff
@@ -16,15 +16,107 @@ import {
   ListPageBody,
   ListPageFilter,
 } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  CheckCircleIcon,
+  ExclamationTriangleIcon,
+  LayerGroupIcon,
+  UploadIcon,
+  OutlinedHourglassIcon,
+} from '@patternfly/react-icons';
 import DropdownWithKebab from './DropdownWithKebab';
 
-const statusConditionsAsString = (obj: any) => {
-  if (!obj.status || !obj.status.conditions) {
-    return '';
+const getStatusLabel = (obj: any) => {
+  // Gateway or HTTPRoute
+  if (obj.kind === 'Gateway' || obj.kind === 'HTTPRoute') {
+    const acceptedCondition = obj.status?.conditions?.find((cond: any) => cond.type === 'Accepted' && cond.status === 'True');
+    const programmedCondition = obj.status?.conditions?.find((cond: any) => cond.type === 'Programmed' && cond.status === 'True');
+    const conflictedCondition = obj.status?.conditions?.find((cond: any) => cond.type === 'Conflicted' && cond.status === 'True');
+    const resolvedRefsCondition = obj.status?.conditions?.find((cond: any) => cond.type === 'ResolvedRefs' && cond.status === 'True');
+    
+    let labelText: string;
+    let color: 'green' | 'blue' | 'red' | 'orange';
+    let icon: React.ReactNode;
+
+    if (acceptedCondition) {
+      labelText = 'Accepted';
+      color = 'green';
+      icon = <CheckCircleIcon />;
+    } else if (programmedCondition) {
+      labelText = 'Programmed';
+      color = 'blue';
+      icon = <CheckCircleIcon />;
+    } else if (conflictedCondition) {
+      labelText = 'Conflicted';
+      color = 'red';
+      icon = <ExclamationTriangleIcon />;
+    } else if (resolvedRefsCondition) {
+      labelText = 'Resolved Refs';
+      color = 'blue';
+      icon = <CheckCircleIcon />;
+    } else {
+      labelText = 'Unknown';
+      color = 'orange';
+      icon = <ExclamationTriangleIcon />;
+    }
+
+    return <Label icon={icon} color={color}>{labelText}</Label>;
   }
-  return obj.status.conditions
-    .map((condition: any) => `${condition.type}=${condition.status}`)
-    .join(',');
+
+  if (!obj.status || !obj.status.conditions || obj.status.conditions.length === 0) {
+    // No status/conditions
+    return (
+      <Label icon={<OutlinedHourglassIcon />} color="cyan">
+        Creating
+      </Label>
+    );
+  }
+  const enforcedCondition = obj.status.conditions.find((cond: any) => cond.type === 'Enforced' && cond.status === 'True');
+  const acceptedCondition = obj.status.conditions.find((cond: any) => cond.type === 'Accepted' && cond.status === 'True');
+  const acceptedConditionFalse = obj.status.conditions.find((cond: any) => cond.type === 'Accepted' && cond.status === 'False');
+  const overriddenCondition = obj.status.conditions.find((cond: any) => cond.type === 'Overridden' && cond.status === 'True');
```

Comment:
I saw the rfc doc with the conditions table and overridden is set to false in it for the policy being overridden. the rest are the same so might be out of date not sure https://docs.kuadrant.io/0.10.0/architecture/rfcs/0004-policy-status/#conditions

Reply:
good spot, updated